Bump golangci-lint to v1.60.2

Let's bump golangci-lint to the latest v1.60.2 release.


Tasks:
* Bump the linter (prior art: https://github.com/kubernetes-sigs/cluster-api/pull/10398/files)
* Run `make lint` and fix new findings



/help

huge warnings about govet printf and gosec G115

```
controlplane/kubeadm/internal/controllers/controller.go:505:176: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(controlPlane.KCP, controlplanev1.CertificatesAvailableCondition, controlplanev1.CertificatesGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                                             ^
controlplane/kubeadm/internal/controllers/helpers.go:201:38: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        clusterv1.ConditionSeverityError, err.Error())
                                                          ^
controlplane/kubeadm/internal/controllers/helpers.go:210:38: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        clusterv1.ConditionSeverityError, err.Error())
                                                          ^
controlplane/kubeadm/internal/controllers/helpers.go:220:39: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                clusterv1.ConditionSeverityError, err.Error())
                                                                  ^
controlplane/kubeadm/internal/controllers/remediation.go:190:161: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                                            ^
controlplane/kubeadm/internal/controllers/remediation.go:224:161: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                                            ^
controlplane/kubeadm/internal/controllers/remediation.go:229:161: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                                            ^
controlplane/kubeadm/internal/controllers/remediation.go:247:159: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                            ^
exp/internal/controllers/machinepool_controller_phases.go:261:141: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(mp, clusterv1.InfrastructureReadyCondition, clusterv1.IncorrectExternalRefReason, clusterv1.ConditionSeverityError, fmt.Sprintf("could not find infra reference of kind %s with name %s", mp.Spec.Template.Spec.InfrastructureRef.Kind, mp.Spec.Template.Spec.InfrastructureRef.Name))
                                                                                                                                                                 ^
cmd/clusterctl/cmd/describe_cluster.go:341:60: printf: non-constant format string in call to (*github.com/fatih/color.Color).Sprintf (govet)
                name = fmt.Sprintf("%s - %s", objectPrefix, gray.Sprintf(name))
                                                                         ^
cmd/clusterctl/cmd/describe_cluster_test.go:142:56: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.UnknownCondition (govet)
                        condition:     conditions.UnknownCondition("C", "", strings.Repeat("s", 150)),
                                                                            ^
controlplane/kubeadm/internal/workload_cluster_conditions.go:446:121: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(machine, staticPodCondition, controlplanev1.PodFailedReason, clusterv1.ConditionSeverityError, strings.Join(containerWaitingMessages, ", "))
                                                                                                                                                    ^
controlplane/kubeadm/internal/workload_cluster_conditions.go:451:125: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(machine, staticPodCondition, controlplanev1.PodProvisioningReason, clusterv1.ConditionSeverityInfo, strings.Join(containerWaitingMessages, ", "))
                                                                                                                                                 ^
controlplane/kubeadm/internal/workload_cluster_conditions.go:463:120: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(machine, staticPodCondition, controlplanev1.PodFailedReason, clusterv1.ConditionSeverityError, strings.Join(containerTerminatedMessages, ", "))
                                                                                                                                            ^
controlplane/kubeadm/internal/workload_cluster_conditions.go:555:122: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(input.controlPlane.KCP, input.condition, input.unhealthyReason, clusterv1.ConditionSeverityError, strings.Join(input.kcpErrors, "; "))
                                                                                                                                       ^
controlplane/kubeadm/internal/workload_cluster_conditions_test.go:1100:116: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                        expectedCondition: *conditions.FalseCondition(conditionType, unhealthyReason, clusterv1.ConditionSeverityError, fmt.Sprintf("Following machines are reporting %s errors: %s", note, "m1")),
                                                                                                                                        ^
controlplane/kubeadm/internal/workload_cluster_conditions_test.go:1115:118: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                        expectedCondition: *conditions.FalseCondition(conditionType, unhealthyReason, clusterv1.ConditionSeverityWarning, fmt.Sprintf("Following machines are reporting %s warnings: %s", note, "m1")),
                                                                                                                                          ^
controlplane/kubeadm/internal/workload_cluster_conditions_test.go:1122:115: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                        expectedCondition: *conditions.FalseCondition(conditionType, unhealthyReason, clusterv1.ConditionSeverityInfo, fmt.Sprintf("Following machines are reporting %s info: %s", note, "m1")),
                                                                                                                                       ^
controlplane/kubeadm/internal/workload_cluster_conditions_test.go:1136:82: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.UnknownCondition (govet)
                        expectedCondition: *conditions.UnknownCondition(conditionType, unknownReason, fmt.Sprintf("Following machines are reporting unknown %s status: %s", note, "m1")),
                                                                                                      ^
util/conditions/getter.go:238:102: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                        condition = FalseCondition(targetCondition, mirrorOpt.fallbackReason, mirrorOpt.fallbackSeverity, mirrorOpt.fallbackMessage)
                                                                                                                          ^
util/conditions/merge.go:79:79: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                return FalseCondition(targetCondition, targetReason, g.TopGroup().severity, targetMessage)
                                                                                            ^
util/conditions/merge.go:81:57: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.UnknownCondition (govet)
        return UnknownCondition(targetCondition, targetReason, targetMessage)
                                                               ^
exp/addons/internal/controllers/clusterresourceset_controller.go:145:151: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.ClusterMatchFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                    ^
exp/addons/internal/controllers/clusterresourceset_controller.go:304:150: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.WrongSecretTypeReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                                 ^
exp/addons/internal/controllers/clusterresourceset_controller.go:306:159: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.RetrievingResourceFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                                          ^
exp/addons/internal/controllers/clusterresourceset_controller.go:358:156: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.RemoteClusterClientFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                         ^
exp/addons/internal/controllers/clusterresourceset_controller.go:409:145: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(clusterResourceSet, addonsv1.ResourcesAppliedCondition, addonsv1.ApplyFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                     ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:505:166: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                                   ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:518:162: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                               ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:524:162: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                               ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:580:157: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesCorruptedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                          ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:584:157: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesCorruptedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                          ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:630:162: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                               ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:636:162: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                               ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:643:163: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                                       ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:709:157: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesCorruptedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                          ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:713:157: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.CertificatesAvailableCondition, bootstrapv1.CertificatesCorruptedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                          ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:747:162: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                               ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:753:162: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                               ^
bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go:760:163: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(scope.Config, bootstrapv1.DataSecretAvailableCondition, bootstrapv1.DataSecretGenerationFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                                       ^
internal/controllers/clusterclass/clusterclass_controller.go:318:5: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                                strings.Join(msg, ", "),
                                ^
internal/controllers/topology/cluster/conditions.go:70:5: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                                reconcileErr.Error(),
                                ^
internal/controllers/topology/cluster/conditions.go:103:5: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                                s.HookResponseTracker.AggregateMessage(),
                                ^
internal/controllers/topology/cluster/conditions.go:200:5: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.FalseCondition (govet)
                                msgBuilder.String(),
                                ^
internal/controllers/machine/machine_controller.go:391:136: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                        conditions.MarkFalse(m, clusterv1.DrainingSucceededCondition, clusterv1.DrainingFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
                                                                                                                                                                          ^
internal/controllers/machine/machine_controller_noderef.go:152:146: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                conditions.MarkFalse(machine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeConditionsFailedReason, clusterv1.ConditionSeverityWarning, message)
                                                                                                                                                               ^
internal/controllers/machine/machine_controller_noderef.go:156:112: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkUnknown (govet)
                conditions.MarkUnknown(machine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeConditionsFailedReason, message)
                                                                                                                             ^
internal/controllers/machineset/machineset_controller.go:475:137: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(ms, clusterv1.MachinesCreatedCondition, clusterv1.PreflightCheckFailedReason, clusterv1.ConditionSeverityError, preflightCheckErrMessage)
                                                                                                                                                             ^
internal/controllers/machineset/machineset_controller.go:512:149: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                        conditions.MarkFalse(ms, clusterv1.MachinesCreatedCondition, clusterv1.BootstrapTemplateCloningFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                                       ^
internal/controllers/machineset/machineset_controller.go:542:153: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                conditions.MarkFalse(ms, clusterv1.MachinesCreatedCondition, clusterv1.InfrastructureTemplateCloningFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                                    ^
internal/controllers/machineset/machineset_controller.go:556:40: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                        clusterv1.ConditionSeverityError, err.Error())
                                                                          ^
internal/controllers/machineset/machineset_controller.go:1107:146: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                        conditions.MarkFalse(m, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason, clusterv1.ConditionSeverityWarning, preflightCheckErrMessage)
                                                                                                                                                                      ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:401:174: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                        conditions.MarkFalse(m, clusterv1.ExternalRemediationTemplateAvailableCondition, clusterv1.ExternalRemediationTemplateNotFoundReason, clusterv1.ConditionSeverityError, err.Error())
                                                                                                                                                                                                                ^
internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:430:178: printf: non-constant format string in call to sigs.k8s.io/cluster-api/util/conditions.MarkFalse (govet)
                                        conditions.MarkFalse(m, clusterv1.ExternalRemediationRequestAvailableCondition, clusterv1.ExternalRemediationRequestCreationFailedReason, clusterv1.ConditionSeverityError, err.Error())
                                                    
```

--- a/.github/workflows/pr-golangci-lint.yaml
+++ b/.github/workflows/pr-golangci-lint.yaml
@@ -30,6 +30,6 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@aaa42aa0628b4ae2578232a66b541047968fac86 # tag=v6.1.0
         with:
-          version: v1.59.0
+          version: v1.60.2
           args: --out-format=colored-line-number
           working-directory: ${{matrix.working-directory}}

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -354,3 +354,8 @@ issues:
     - gocritic
     text: "deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop"
     path: _test\.go
+  exclude-files:
+    - "zz_generated.*\\.go$"
+    - "vendored_openapi\\.go$"
+    # We don't want to invest time to fix new linter findings in old API types.
+    - "internal/apis/.*"

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -55,11 +55,9 @@ linters:
     - whitespace # unnecessary newlines
 
 linters-settings:
-  govet:
-    disable:
-      - printf
   gosec:
     excludes:
+    # integer overflow conversion int -> int32
       - G115
   gci:
     sections:
@@ -354,6 +352,10 @@ issues:
     - gocritic
     text: "deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop"
     path: _test\.go
+    # Ignore non-constant format string in call to condition utils
+  - linters:
+      - govet
+    text: "non-constant format string in call to sigs\\.k8s\\.io\\/cluster-api\\/util\\/conditions\\."
   exclude-files:
     - "zz_generated.*\\.go$"
     - "vendored_openapi\\.go$"

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -55,6 +55,12 @@ linters:
     - whitespace # unnecessary newlines
 
 linters-settings:
+  govet:
+    disable:
+      - printf
+  gosec:
+    excludes:
+      - G115
   gci:
     sections:
       - standard # Standard section: captures all standard packages.

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -356,8 +356,3 @@ issues:
   - linters:
       - govet
     text: "non-constant format string in call to sigs\\.k8s\\.io\\/cluster-api\\/util\\/conditions\\."
-  exclude-files:
-    - "zz_generated.*\\.go$"
-    - "vendored_openapi\\.go$"
-    # We don't want to invest time to fix new linter findings in old API types.
-    - "internal/apis/.*"

--- a/cmd/clusterctl/cmd/describe_cluster.go
+++ b/cmd/clusterctl/cmd/describe_cluster.go
@@ -338,7 +338,7 @@ func getRowName(obj ctrlclient.Object) string {
 
 	name := objName
 	if objectPrefix := tree.GetMetaName(obj); objectPrefix != "" {
-		name = fmt.Sprintf("%s - %s", objectPrefix, gray.Sprintf(name))
+		name = fmt.Sprintf("%s - %s", objectPrefix, gray.Sprintf("%s", name))
 	}
 
 	if !obj.GetDeletionTimestamp().IsZero() {

--- a/internal/controllers/machinedeployment/mdutil/util.go
+++ b/internal/controllers/machinedeployment/mdutil/util.go
@@ -120,16 +120,16 @@ func SetDeploymentRevision(deployment *clusterv1.MachineDeployment, revision str
 func MaxRevision(ctx context.Context, allMSs []*clusterv1.MachineSet) int64 {
 	log := ctrl.LoggerFrom(ctx)
 
-	max := int64(0)
+	maxVal := int64(0)
 	for _, ms := range allMSs {
 		if v, err := Revision(ms); err != nil {
 			// Skip the machine sets when it failed to parse their revision information
 			log.Error(err, fmt.Sprintf("Couldn't parse revision for MachineSet %s, deployment controller will skip it when reconciling revisions", ms.Name))
-		} else if v > max {
-			max = v
+		} else if v > maxVal {
+			maxVal = v
 		}
 	}
-	return max
+	return maxVal
 }
 
 // Revision returns the revision number of the input object.

--- a/internal/controllers/machinehealthcheck/machinehealthcheck_controller.go
+++ b/internal/controllers/machinehealthcheck/machinehealthcheck_controller.go
@@ -580,13 +580,13 @@ func isAllowedRemediation(mhc *clusterv1.MachineHealthCheck) (bool, int32, error
 	var remediationAllowed bool
 	var remediationCount int32
 	if mhc.Spec.UnhealthyRange != nil {
-		min, max, err := getUnhealthyRange(mhc)
+		minVal, maxVal, err := getUnhealthyRange(mhc)
 		if err != nil {
 			return false, 0, err
 		}
 		unhealthyMachineCount := unhealthyMachineCount(mhc)
-		remediationAllowed = unhealthyMachineCount >= min && unhealthyMachineCount <= max
-		remediationCount = int32(max - unhealthyMachineCount)
+		remediationAllowed = unhealthyMachineCount >= minVal && unhealthyMachineCount <= maxVal
+		remediationCount = int32(maxVal - unhealthyMachineCount)
 		return remediationAllowed, remediationCount, nil
 	}
 
@@ -610,21 +610,21 @@ func getUnhealthyRange(mhc *clusterv1.MachineHealthCheck) (int, int, error) {
 
 	parts := strings.Split(unhealthyRange, "-")
 
-	min, err := strconv.ParseUint(parts[0], 10, 32)
+	minVal, err := strconv.ParseUint(parts[0], 10, 32)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	max, err := strconv.ParseUint(parts[1], 10, 32)
+	maxVal, err := strconv.ParseUint(parts[1], 10, 32)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	if max < min {
-		return 0, 0, errors.Errorf("max value %d cannot be less than min value %d for unhealthyRange", max, min)
+	if maxVal < minVal {
+		return 0, 0, errors.Errorf("max value %d cannot be less than min value %d for unhealthyRange", maxVal, minVal)
 	}
 
-	return int(min), int(max), nil
+	return int(minVal), int(maxVal), nil
 }
 
 func getMaxUnhealthy(mhc *clusterv1.MachineHealthCheck) (int, error) {

--- a/internal/controllers/machinehealthcheck/machinehealthcheck_targets_test.go
+++ b/internal/controllers/machinehealthcheck/machinehealthcheck_targets_test.go
@@ -406,7 +406,7 @@ func TestHealthCheckTargets(t *testing.T) {
 	machineFailureMsgCondition := newFailedHealthCheckCondition(clusterv1.MachineHasFailureReason, "FailureMessage: %s", failureMsg)
 
 	// Target for when the machine has the remediate machine annotation
-	annotationRemediationMsg := "Marked for remediation via remediate-machine annotation"
+	const annotationRemediationMsg = "Marked for remediation via remediate-machine annotation"
 	testMachineAnnotationRemediation := testMachine.DeepCopy()
 	testMachineAnnotationRemediation.Annotations = map[string]string{clusterv1.RemediateMachineAnnotation: ""}
 	machineAnnotationRemediation := healthCheckTarget{

--- a/test/framework/docker_logcollector.go
+++ b/test/framework/docker_logcollector.go
@@ -147,7 +147,7 @@ func (k DockerLogCollector) collectLogsFromNode(ctx context.Context, outputPath 
 				"tar", "--hard-dereference", "--dereference", "--directory", containerDir, "--create", "--file", "-", ".",
 			)
 			if err != nil {
-				return errors.Wrapf(err, execErr)
+				return errors.Wrap(err, execErr)
 			}
 
 			err = os.MkdirAll(outputDir, 0750)

--- a/test/infrastructure/docker/internal/docker/loadbalancer.go
+++ b/test/infrastructure/docker/internal/docker/loadbalancer.go
@@ -65,7 +65,10 @@ func NewLoadBalancer(ctx context.Context, cluster *clusterv1.Cluster, dockerClus
 		return nil, err
 	}
 
-	ipFamily, err := cluster.GetIPFamily() //nolint:staticcheck // We tolerate this until removal; after removal IPFamily will become an internal CAPD concept. See https://github.com/kubernetes-sigs/cluster-api/issues/7521.
+	// We tolerate this until removal;
+	// after removal IPFamily will become an internal CAPD concept.
+	// See https://github.com/kubernetes-sigs/cluster-api/issues/7521.
+	ipFamily, err := cluster.GetIPFamily() //nolint:staticcheck
 	if err != nil {
 		return nil, fmt.Errorf("create load balancer: %s", err)
 	}

--- a/test/infrastructure/docker/internal/docker/machine.go
+++ b/test/infrastructure/docker/internal/docker/machine.go
@@ -92,7 +92,10 @@ func NewMachine(ctx context.Context, cluster *clusterv1.Cluster, machine string,
 		return nil, err
 	}
 
-	ipFamily, err := cluster.GetIPFamily() //nolint:staticcheck // We tolerate this until removal; after removal IPFamily will become an internal CAPD concept. See https://github.com/kubernetes-sigs/cluster-api/issues/7521.
+	// We tolerate this until removal;
+	// after removal IPFamily will become an internal CAPD concept.
+	// See https://github.com/kubernetes-sigs/cluster-api/issues/7521.
+	ipFamily, err := cluster.GetIPFamily() //nolint:staticcheck
 	if err != nil {
 		return nil, fmt.Errorf("create docker machine: %s", err)
 	}
@@ -126,7 +129,10 @@ func ListMachinesByCluster(ctx context.Context, cluster *clusterv1.Cluster, labe
 		return nil, err
 	}
 
-	ipFamily, err := cluster.GetIPFamily() //nolint:staticcheck // We tolerate this until removal; after removal IPFamily will become an internal CAPD concept. See https://github.com/kubernetes-sigs/cluster-api/issues/7521.
+	// We tolerate this until removal;
+	// after removal IPFamily will become an internal CAPD concept.
+	// See https://github.com/kubernetes-sigs/cluster-api/issues/7521 .
+	ipFamily, err := cluster.GetIPFamily() //nolint:staticcheck
 	if err != nil {
 		return nil, fmt.Errorf("list docker machines by cluster: %s", err)
 	}

--- a/test/infrastructure/inmemory/pkg/runtime/cache/client.go
+++ b/test/infrastructure/inmemory/pkg/runtime/cache/client.go
@@ -212,9 +212,8 @@ func (c *cache) store(resourceGroup string, obj client.Object, replaceExisting b
 				return apierrors.NewConflict(unsafeGuessGroupVersionResource(objGVK).GroupResource(), objKey.String(), fmt.Errorf("object has been modified"))
 			}
 
-			if err := c.beforeUpdate(resourceGroup, trackedObj, obj); err != nil {
-				return err
-			}
+			c.beforeUpdate(resourceGroup, trackedObj, obj)
+
 			tracker.objects[objGVK][objKey] = obj.DeepCopyObject().(client.Object)
 			updateTrackerOwnerReferences(tracker, trackedObj, obj, objRef)
 			c.afterUpdate(resourceGroup, trackedObj, obj)
@@ -227,9 +226,8 @@ func (c *cache) store(resourceGroup string, obj client.Object, replaceExisting b
 		return apierrors.NewNotFound(unsafeGuessGroupVersionResource(objGVK).GroupResource(), objKey.String())
 	}
 
-	if err := c.beforeCreate(resourceGroup, obj); err != nil {
-		return err
-	}
+	c.beforeCreate(resourceGroup, obj)
+
 	tracker.objects[objGVK][objKey] = obj.DeepCopyObject().(client.Object)
 	updateTrackerOwnerReferences(tracker, nil, obj, objRef)
 	c.afterCreate(resourceGroup, obj)
@@ -424,9 +422,7 @@ func (c *cache) doTryDeleteLocked(resourceGroup string, tracker *resourceGroupTr
 		oldObj := obj.DeepCopyObject().(client.Object)
 		now := metav1.Time{Time: time.Now().UTC()}
 		obj.SetDeletionTimestamp(&now)
-		if err := c.beforeUpdate(resourceGroup, oldObj, obj); err != nil {
-			return false, apierrors.NewBadRequest(err.Error())
-		}
+		c.beforeUpdate(resourceGroup, oldObj, obj)
 
 		objects[objKey] = obj
 		c.afterUpdate(resourceGroup, oldObj, obj)

--- a/test/infrastructure/inmemory/pkg/runtime/cache/hooks.go
+++ b/test/infrastructure/inmemory/pkg/runtime/cache/hooks.go
@@ -27,20 +27,19 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (c *cache) beforeCreate(_ string, obj client.Object) error {
+func (c *cache) beforeCreate(_ string, obj client.Object) {
 	now := time.Now().UTC()
 	obj.SetCreationTimestamp(metav1.Time{Time: now})
 	// TODO: UID
 	obj.SetAnnotations(appendAnnotations(obj, lastSyncTimeAnnotation, now.Format(time.RFC3339)))
 	obj.SetResourceVersion(fmt.Sprintf("v%d", 1))
-	return nil
 }
 
 func (c *cache) afterCreate(resourceGroup string, obj client.Object) {
 	c.informCreate(resourceGroup, obj)
 }
 
-func (c *cache) beforeUpdate(_ string, oldObj, newObj client.Object) error {
+func (c *cache) beforeUpdate(_ string, oldObj, newObj client.Object) {
 	newObj.SetCreationTimestamp(oldObj.GetCreationTimestamp())
 	newObj.SetResourceVersion(oldObj.GetResourceVersion())
 	// TODO: UID
@@ -55,7 +54,6 @@ func (c *cache) beforeUpdate(_ string, oldObj, newObj client.Object) error {
 		oldResourceVersion, _ := strconv.Atoi(strings.TrimPrefix(oldObj.GetResourceVersion(), "v"))
 		newObj.SetResourceVersion(fmt.Sprintf("v%d", oldResourceVersion+1))
 	}
-	return nil
 }
 
 func (c *cache) afterUpdate(resourceGroup string, oldObj, newObj client.Object) {

